controllers/nodefeaturediscovery_status.go

```diff
@@ -0,0 +1,551 @@
+	conditionFailedGettingNFDWorkerDaemonSet      = "FailedGettingNFDWorkerDaemonSet"
+	conditionFailedGettingNFDMasterDaemonSet      = "FailedGettingNFDMasterDaemonSet"
+	conditionFailedGettingNFDRoleBinding          = "FailedGettingNFDRoleBinding"
+
+	// Resource degraded
+	conditionNFDWorkerConfigDegraded         = "NFDWorkerConfigResourceDegraded"
+	conditionNFDWorkerServiceAccountDegraded = "NFDWorkerServiceAccountDegraded"
+	conditionNFDMasterServiceAccountDegraded = "NFDMasterServiceAccountDegraded"
+	conditionNFDServiceDegraded              = "NFDServiceDegraded"
+	conditionNFDWorkerDaemonSetDegraded      = "NFDWorkerDaemonSetDegraded"
+	conditionNFDMasterDaemonSetDegraded      = "NFDMasterDaemonSetDegraded"
+	conditionNFDRoleDegraded                 = "NFDRoleDegraded"
+	conditionNFDRoleBindingDegraded          = "NFDRoleBindingDegraded"
+	conditionNFDClusterRoleDegraded          = "NFDClusterRoleDegraded"
+	conditionNFDClusterRoleBindingDegraded   = "NFDClusterRoleBindingDegraded"
+
+	// Unknown errors. (Catch all)
+	errorNFDWorkerDaemonSetUnknown = "NFDWorkerDaemonSetCorrupted"
+	errorNFDMasterDaemonSetUnknown = "NFDMasterDaemonSetCorrupted"
+
+	// Invalid node type. (Denotes that the node should be either
+	// 'worker' or 'master')
+	errorInvalidNodeType = "InvalidNodeTypeSelected"
+
+	// More nodes are listed as "ready" than selected
+	errorTooManyNFDWorkerDaemonSetReadyNodes = "NFDWorkerDaemonSetHasMoreNodesThanScheduled"
+	errorTooManyNFDMasterDaemonSetReadyNodes = "NFDMasterDaemonSetHasMoreNodesThanScheduled"
+
+	// DaemonSet warnings (for "Progressing" conditions)
+	warningNumberOfReadyNodesIsLessThanScheduled = "warningNumberOfReadyNodesIsLessThanScheduled"
+	warningNFDWorkerDaemonSetProgressing         = "warningNFDWorkerDaemonSetProgressing"
+	warningNFDMasterDaemonSetProgressing         = "warningNFDMasterDaemonSetProgressing"
+
+	// ConditionAvailable indicates that the resources maintained by the operator,
+	// is functional and available in the cluster.
+	ConditionAvailable string = "Available"
+
+	// ConditionProgressing indicates that the operator is actively making changes to the resources maintained by the
+	// operator
+	ConditionProgressing string = "Progressing"
+
+	// ConditionDegraded indicates that the resources maintained by the operator are not functioning completely.
+	// An example of a degraded state would be if not all pods in a deployment were running.
+	// It may still be available, but it is degraded
+	ConditionDegraded string = "Degraded"
+
+	// ConditionUpgradeable indicates whether the resources maintained by the operator are in a state that is safe to upgrade.
+	// When `False`, the resources maintained by the operator should not be upgraded and the
+	// message field should contain a human readable description of what the administrator should do to
+	// allow the operator to successfully update the resources maintained by the operator.
+	ConditionUpgradeable string = "Upgradeable"
```

Comment:
Hmm, do we need this? `ConditionUpgradeable` always has the same status as `ConditionAvailable` so it looks redundant to me 🤔  Could it be dropped, and, added in the future if a need arises?

Reply:
Is needed for managed services, this was a recommendation from SRE teams so they can easly know if they can do upgrades

Reply:
K, fair enough 👍